adapt to GHC-8.0.1

Using `ghc-8.0.0.20160109` I get:
~~~~
Preprocessing library reactive-banana-1.1.0.0...
[ 1 of 22] Compiling Reactive.Banana.Prim.Util ( src/Reactive/Banana/Prim/Util.hs, dist/build/Reactive/Banana/Prim/Util.o )

src/Reactive/Banana/Prim/Util.hs:54:25: error:
    • Couldn't match expected type ‘GHC.State# GHC.RealWorld
                                    -> (# GHC.State# GHC.RealWorld, c0 #)’
                  with actual type ‘IO ()’
    • In the third argument of ‘GHC.mkWeak#’, namely ‘f’
      In the expression: GHC.mkWeak# r# v f s
      In the expression:
        case GHC.mkWeak# r# v f s of {
          (# s1, w #) -> (# s1, GHC.Weak w #) }
Failed to install reactive-banana-1.1.0.0
~~~~

--- a/reactive-banana/src/Reactive/Banana/Prim/Util.hs
+++ b/reactive-banana/src/Reactive/Banana/Prim/Util.hs
@@ -1,3 +1,4 @@
+{-# LANGUAGE CPP #-}
 {-----------------------------------------------------------------------------
     reactive-banana
 ------------------------------------------------------------------------------}
@@ -50,8 +51,13 @@ modify' ~(Ref ref _) f = liftIO $ readIORef ref >>= \x -> writeIORef ref $! f x
     Weak pointers
 ------------------------------------------------------------------------------}
 mkWeakIORefValueFinalizer :: IORef a -> value -> IO () -> IO (Weak value)
+#if MIN_VERSION_base(4,9,0)
 mkWeakIORefValueFinalizer r@(GHC.IORef (GHC.STRef r#)) v (GHC.IO f) = GHC.IO $ \s ->
   case GHC.mkWeak# r# v f s of (# s1, w #) -> (# s1, GHC.Weak w #)
+#else
+mkWeakIORefValueFinalizer r@(GHC.IORef (GHC.STRef r#)) v f = GHC.IO $ \s ->
+  case GHC.mkWeak# r# v f s of (# s1, w #) -> (# s1, GHC.Weak w #)
+#endif
 
 mkWeakIORefValue :: IORef a -> value -> IO (Weak value)
 mkWeakIORefValue a b = mkWeakIORefValueFinalizer a b (return ())

--- a/reactive-banana/src/Reactive/Banana/Prim/Util.hs
+++ b/reactive-banana/src/Reactive/Banana/Prim/Util.hs
@@ -50,7 +50,7 @@ modify' ~(Ref ref _) f = liftIO $ readIORef ref >>= \x -> writeIORef ref $! f x
     Weak pointers
 ------------------------------------------------------------------------------}
 mkWeakIORefValueFinalizer :: IORef a -> value -> IO () -> IO (Weak value)
-mkWeakIORefValueFinalizer r@(GHC.IORef (GHC.STRef r#)) v f = GHC.IO $ \s ->
+mkWeakIORefValueFinalizer r@(GHC.IORef (GHC.STRef r#)) v (GHC.IO f) = GHC.IO $ \s ->
   case GHC.mkWeak# r# v f s of (# s1, w #) -> (# s1, GHC.Weak w #)
 
 mkWeakIORefValue :: IORef a -> value -> IO (Weak value)

